src/components/globalConfigurations/GlobalConfiguration.tsx

```diff
@@ -200,26 +200,42 @@ function NavItem({ hostURLConfig, serverMode }) {
         !location.pathname.includes(URLS.GLOBAL_CONFIG_HOST_URL)
 
     useEffect(() => {
-        getGitOpsModuleStatus()
-        getNotificationModuleStatus()
+      getGitOpsModuleStatus(3)
+      getNotificationModuleStatus(3)
     }, [])
 
-    async function getGitOpsModuleStatus() {
+    const getGitOpsModuleStatus = async (retryOnError: number): Promise<void> => {
         try {
             const { result } = await getModuleInfo(ModuleNameMap.ARGO_CD)
             if (result?.status === ModuleStatus.INSTALLED) {
                 setInstalledModule([...installedModule, ModuleNameMap.ARGO_CD])
+            } else if (result?.status === ModuleStatus.INSTALLING) {
+                moduleStatusTimer = setTimeout(() => {
+                    getGitOpsModuleStatus(3)
+                }, 15000)
+            }
+        } catch (error) {
+            if (retryOnError >= 0) {
+                getGitOpsModuleStatus(retryOnError--)
             }
-        } catch (error) {}
+        }
     }
 
-    const getNotificationModuleStatus = async (): Promise<void> => {
+    const getNotificationModuleStatus = async (retryOnError: number): Promise<void> => {
         try {
             const { result } = await getModuleInfo(ModuleNameMap.NOTIFICATION)
             if (result?.status === ModuleStatus.INSTALLED) {
                 setInstalledModule([...installedModule, ModuleNameMap.NOTIFICATION])
+            } else if (result?.status === ModuleStatus.INSTALLING) {
+                moduleStatusTimer = setTimeout(() => {
+                  getNotificationModuleStatus(3)
+                }, 15000)
             }
-        } catch (error) {}
+        } catch (error) {
+            if (retryOnError >= 0) {
```

Comment:
Will there be any condition where `retryOnError < 0` if yes how are we handling that

Reply:
No